plugin/src/ast_helpers.rs

```diff
@@ -650,7 +650,7 @@ fn is_new_line(ch: char) -> bool {
 pub fn parse(source_text: &str) -> Tree {
     let mut parser = Parser::new();
     parser
-        .set_language(SupportedLanguage::Javascript.language())
+        .set_language(SupportedLanguage::Javascript.language(None))
```

Comment:
To handle the fact that Typescript maps to two different tree-sitter grammars/`Language`'s, I introduced an intermediate `SupportedLanguageLanguage` layer (so a `SupportedLanguage` maps to one or more `SupportedLanguageLanguage`'s (and you can map to just one by supplying an `Option<&Path>`, so basically disambiguating based on file extension + a "default" fallback), and a `SupportedLanguageLanguage` maps 1-to-1 to a `tree_sitter::Language`)